### ISSUE TITLE
Re-licensing of pdfpc to GPLv3

### ISSUE THREAD

**Issue by AndreasBilke** [pdfpc/pdfpc#499]
We are planning to re-license pdfpc to GPLv3. Currently, we have a GPLv2+ license because in 2011 there was a license issue with the poppler library which forced the former project maintainer to that step.
Since the end of 2011 poppler has a GPLv2+ license which, in our opinion, could allow us to go to GPLv3 directly.
Because @barak pointed out the license problem back then I asked him via email for guidance if we could to this step now. He asked me if we can make this conversation public.

Here we are. :-)

TLDR: Can we re-license pdfpc to GPLv3 now?

**Comment by LecrisUT** [comment]
Could you clarify what the (main) difference between the two GPL licenses are?

**Comment by fnevgeny** [comment]
https://www.gnu.org/licenses/rms-why-gplv3.html

**Comment by fnevgeny** [comment]
Well, I think we should proceed...

**Comment by barak** [comment]
I'm assuming when you say switch from GPLv2+ to GPLv3 you mean to GPLv3+ (although there's no GPLv4 and no signs of one, so it makes no practical difference for the near future.)

Since GPLv2+ ⊃ GPLv3+, I don't think you *need* to switch. It seems to me that there are really only a few reason to switch from GPLv2+ to GPLv3+:

- there's something GPLv2 permits people to do that you would like to forbid
- there's some code under GPLv3+ and not GPLv2 that you'd like to incorporate

Is there some particular issue that's come up? You can always switch, but it's hard to switch back because you'd need to get permission from anyone who's sent in code under the understanding that it was GPLv3+.

**Comment by fnevgeny** [comment]
Hi @barak ,

Yes, we are talking about GPLv3+.

Which reasons might make us think of switching back to GPLv2+?

Note that even now, we cannot incorporate a *strict* GPLv2 code; and GPLv2+ shouldn't be a problem to use in GPLv3+. Am I wrong?

**Comment by barak** [comment]
No, that sounds right.

I guess I'm thinking about the following scenario.

You incorporate `foo.vala` which is GPLv2+. There are some bugs in it, which you fix, you add some new foo-ish features, etc, resulting in a greatly improved, but still modular and coherent, `foo.vala`. Then whoever you snagged `foo.vala` from wants to pull back your improvements. But the rest of their codebase is GPLv2+ because they're linking to a GPLv2-only library or something like that. So if your changes were GPLv2+ they could pull in your stuff, but if they were GPLv3+ then they couldn't.

**Comment by fnevgeny** [comment]
Well, then it's the problem of the authors of that `foo.vala`, not ours :). Not that I like to make somebody's life harder than necessary, but the chances of such a collision seems to me pretty low.

On the other hand, it probably makes no sense to make licensing changes at this point (quite close to the 4.5 release), where all of the post-4.4 development has still been done under GPLv2+. I propose instead to do it as the **first** step towards 4.6, once 4.5 is released.
